plugin-server/src/worker/ingestion/event-pipeline/extractHeatmapDataStep.ts

```diff
@@ -57,13 +58,20 @@ function replacePathInUrl(url: string, newPath: string): string {
     return parsedUrl.toString()
 }
 
+function isValidString(x: unknown): x is string {
+    return typeof x === 'string' && !!x.trim().length
+}
+
+function isValidNumber(n: unknown): n is number {
+    return typeof n === 'number' && !isNaN(n)
+}
+
 function extractScrollDepthHeatmapData(event: PreIngestionEvent): RawClickhouseHeatmapEvent[] {
-    const { teamId, timestamp, properties } = event
+    const { teamId, timestamp, properties, distinctId } = event
```

Comment:
this is actually the bug, we were reading distinct id from event properties and that's not sanitized earlier in the pipeline